### PR TITLE
Add support for passthrough Elasticsearch queries

### PR DESCRIPTION
**Pull request by v-jizhang**
Cherry-pick of https://github.com/trinodb/trino/pull/3735
This allows running queries over the results of a raw Elasticsearch query.
It extends the syntax of the enhanced ES table names with the following:

SELECT * FROM es.default."<index>$query:<base32-encoded ES query>"

The query is base32-encoded to avoid having to deal with escaping quotes and case
sensitivity issues in table identifiers.

The result of these query tables is a table with a single row and a single column
named "result" of type JSON.

Co-authored-by: Martin Traverso <mtraverso@gmail.com>

```
== RELEASE NOTES ==

General Changes
* Add support for passthrough Elasticsearch queries
```

### CODE REVIEW

**Comment by aweisberg** [comment]
The Trino PR says it depends on https://github.com/trinodb/trino/pull/3718/files, but it looks to me like we already have it?
@zhenxiao tagging you also for review because you have done a lot of work in this area.

**Comment by aweisberg** [comment]
One note, we should also bring the elastic search connector doc up to date trinodb/trino@f776741

**Comment by v-jizhang** [comment]
> The Trino PR says it depends on https://github.com/trinodb/trino/pull/3718/files, but it looks to me like we already have it?
> @zhenxiao tagging you also for review because you have done a lot of work in this area.

Yes, we have it already.

**Comment by v-jizhang** [comment]
> One note, we should also bring the elastic search connector doc up to date [trinodb/trino@f776741](https://github.com/trinodb/trino/commit/f776741)

I'll add it.

File: presto-spi/src/main/java/com/facebook/presto/spi/ColumnMetadata.java [hunk 1]
```diff
@@ -151,4 +172,76 @@ public boolean equals(Object obj)
                 Objects.equals(this.extraInfo, other.extraInfo) &&
                 Objects.equals(this.hidden, other.hidden);
     }
+
```
Comment (aweisberg) [inline presto-spi/src/main/java/com/facebook/presto/spi/ColumnMetadata.java:175]: This isn't all of https://github.com/trinodb/trino/commit/6415b9aa9f5df0713389e2d4c87cfa71c4355cde
It makes sense as a separate commit more.

File: presto-spi/src/main/java/com/facebook/presto/spi/StandardErrorCode.java [hunk 2]
```diff
@@ -65,6 +65,7 @@
     INVALID_ANALYZE_PROPERTY(0x0000_002A, USER_ERROR),
     GENERATED_BYTECODE_TOO_LARGE(0x0000_002B, USER_ERROR),
     WARNING_AS_ERROR(0x0000_002C, USER_ERROR),
+    INVALID_ARGUMENTS(0x0000_002D, USER_ERROR),
```
Comment (aweisberg) [inline presto-spi/src/main/java/com/facebook/presto/spi/StandardErrorCode.java:68]: Hmm we don't have https://github.com/martint/trino/commit/1f40322942c92c50a410c65f42f36f0629d2d4fb so it's questionable whether this makes sense in isolation or if we should use some other existing error code.

Martin's reasoning seems sound. @arhimondr  @rongrong  @kaikalur do you have any opinions on direction for error codes?

**Review by aweisberg — CHANGES_REQUESTED** [review]
This looks good to me modulo the doc changes and the unresolved question about that error code. Whether we should take all the semantic error code removal changes or whether there is a different better error code to use in this case.

Pulling up to a higher level I was a little surprised the overlap in functionality between plain text pass through queries and base32 passthrough queries. I wonder what drove the introduction of base32 encoded passthrough.

This also highlights that we are missing limit and filter pushdown for ES and that isn't in the list of pending improvements I see.

**Comment by v-jizhang** [comment]
> The Trino PR says it depends on https://github.com/trinodb/trino/pull/3718/files, but it looks to me like we already have it?
> @zhenxiao tagging you also for review because you have done a lot of work in this area.

Yes, we ready have it.

File: presto-spi/src/main/java/com/facebook/presto/spi/ColumnMetadata.java (hunk 1 above)
Comment (v-jizhang) [reply presto-spi/src/main/java/com/facebook/presto/spi/ColumnMetadata.java:175]: Yeah, I'll commit separately next time. Thanks

**Comment by v-jizhang** [comment]
The failed test is not related. Looks like a network problem on CI.

**Review by aweisberg — APPROVED** [review]
Documentation commit needs a co-author and link to the source. Can you also squash and clean up the history?

**Comment by v-jizhang** [comment]
> Documentation commit needs a co-author and link to the source. Can you also squash and clean up the history?

Added co-author, thanks.

**Comment by aweisberg** [comment]
@zhenxiao do you have time to review/merge this?

**Review by zhenxiao — APPROVED** [review]
looks good

**Pull request merged**